honeybee_vtk/writer.py

```diff
@@ -0,0 +1,141 @@
+        raise ValueError(
+            'Not a valid JSON file.'
+            )
+    else:
+        # Get points and face_types from HBJSON
+        points_lst, hb_types = read_hbjson(hbjson)
+
+        # Get points grouped for each Honeybee face_type
+        grouped_points = group_by_face_type(points_lst, hb_types)
+
+        # Write VTK files based on Honeybee face_type and Honeybee object type
+        for hb_type in grouped_points:
+            write_polydata(grouped_points[hb_type], hb_type)
+
+        # Names of VTK files to be written
+        file_names = list(grouped_points.keys())
+
+        # write grid points to a vtk file if grids are found in HBJSON
+        grids = check_grid(hbjson)
+
+        # If grids are there in HBJSON and they are requested
+        if grids and get_grids:
+
+            # If base_geometry is found
+            if grids[0]:
+                base_geo_points = get_grid_base(grids[0])[0]
+                write_polydata(base_geo_points, 'grid base')
+                file_names.append('grid base')
+
+            # If base_geometry is not found but mesh faces are found
+            if grids[1]:
+                mesh_points = get_grid_mesh(grids[1])[0]
+                write_polydata(mesh_points, 'grid mesh')
+                file_names.append('grid mesh')
+
+            # If only grid points are found
+            if grids[2]:
+                start_points, end_points, vectors = get_grid_points(grids[2])
+                write_color_grouped_points(start_points, vectors)
+                file_names.append('grid points')
+        else:
+            pass
+        
+        # Write vectors if they are requested
+        if get_vectors:
+            
+            # If Aperture objects are there in HBJSON
+            if 'Aperture' in hb_types:
+                # Create face normals
+                start_points, end_points, normals = get_face_center(
+                    grouped_points['Aperture'])
+                write_arrows(start_points, end_points, normals, 'Aperture')
+                file_names.append('Aperture vectors')
+            
+            # If grids are found in HBJSON
+            if grids:
+                
+                # If base_geometry is found in any of the grids
+                if grids[0]:
+                    base_geo_points = get_grid_base(grids[0])[0]
+                    start_points, end_points, normals = get_face_center(base_geo_points)
+                    write_arrows(start_points, end_points, normals, 'grid base')
+                    file_names.append('grid base vectors')
+
+                # If mesh is found in any of the grids
+                if grids[1]:
+                    mesh_points = get_grid_mesh(grids[1])[0]
+                    start_points, end_points, normals = get_face_center(mesh_points)
+                    write_arrows(start_points, end_points, normals, 'grid mesh')
+                    file_names.append('grid mesh vectors')
+        else:
+            pass
```

Comment:
I would move each of these `if` clauses to a separate private function in this module.

if get_vectors:
    _get_vectors(..., )